redhat-appstudio/service-provider-integration-operator
# Add Common username + token provider

### What does this PR do?

As it is supposed that Snyk is not the last provider to support `username` + `token` only manual upload ability, it was deciced to do that provider universal, i..e support any URL-s and perform tokens matching by them. One can still provide attitional info during upload, but it will not be taken into account during matching. So the prinicple is `single token per namespace per host`.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
[]
(https://issues.redhat.com/browse/SVPI-118)


### How to test this PR?
 - Build image;
 - Create SPIAccessTokenBinding for any URL;
 Example:
 ```
apiVersion: appstudio.redhat.com/v1beta1
kind: SPIAccessTokenBinding
metadata:
  name: test-access-token-binding-2
  namespace: default
spec:
  repoUrl: https://snyk.io/
  secret:
    type: kubernetes.io/basic-auth

 ```
 - Check binding created and grab linked SPIAccessToken name;
 - Perform usual manual token uploading, containing at least username + token;
 
 ```
curl -k -v -X POST -H "Content-Type: application/json" -H "Authorization: Bearer $(hack/get-default-sa-token.sh)" -d'{"username": "uy7c+spi_demo", "access_token": "4R28N79MT"}' "http://spi.$(minikube ip).nip.io/token/default/generated-spi-access-token-5sp5c"
 ```
 - Make sure token is `Ready`, secrets created, etc
 - Create another binding to the same URL. It should match to existing token immediately.
 - Create another binding with different URL. New SPIAccessToken with `AwaitingToken` state must be created.
